core/src/main/java/de/danoeh/antennapod/core/feed/FeedMedia.java

```diff
@@ -167,6 +168,11 @@ public MediaType getMediaType() {
     }
 
     public void updateFromOther(FeedMedia other) {
+        // we try to cover two cases: (1) feed did include file before (2) feed contained wrong URL
+        // if item.getAutoDownload() is false, the file has been downloaded before
+        if((TextUtils.isEmpty(download_url) || item.getAutoDownload()) && !TextUtils.isEmpty(other.download_url)) {
+            item.setNew();
```

Comment:
The idea is to also reset the state if the item already had a URL, but the URL was not valid (404'd, had a typo, etc.). ``getAutoDownload`` should only be ``0`` if the URL was wrong for at least 2 weeks (exp. back-off)
We could probably check the logs (if not cleared in the meantime), but that would be rather cumbersome.